configure.ac

```diff
@@ -132,14 +132,17 @@ AS_IF([test x"$with_rpc" != "xno"], [
             [no], [break],
             [*], [PKG_CHECK_MODULES([RPC], ["$rpc_impl"], [], [continue])])
         have_rpc=maybe
+        save_CFLAGS="$CFLAGS"
+        CFLAGS="$CFLAGS $RPC_CFLAGS"
         AC_CHECK_HEADERS([rpc/rpc.h],, [have_rpc=no; break])
         dnl FIXME do a linking test
         AS_IF([test x"$have_rpc" = xmaybe], [
             dnl XXX is this header really optional?
             AC_CHECK_HEADERS([rpc/rpcent.h])
             have_rpc=yes
-            break
         ])
+        CFLAGS="$save_CFLAGS"
+        test x"$have_rpc" = xyes && break
```

Comment:
use `AS_IF` here too please

Reply:
any reason for using `if` there?

Reply:
`AS_IF` provides more structure to the code, and avoids unintended side-effects with `&&` & `||` usage.  it lends itself to more robust code without having to think as hard about it.